### PR TITLE
447 achieve 100 line coverage on public api

### PR DESCRIPTION
**Pull request by ComLock**

### CODE REVIEW

File: src/i18n/LocaleContext.tsx [hunk 1]
```diff
@@ -29,16 +34,22 @@ export const LocaleContextProvider = ({children, locale}: { children: any, local
         });
     };
 
-    if (locale && currentLocale !== locale) {
-        // async/await is not supported in client components yet
-        void setLocale(locale);
-    }
+    useEffect(() => {
+        if (localeProps) {
+            setLocale(localeProps)
```
Comment (pmi) [inline src/i18n/LocaleContext.tsx:39]: I'd use `void setLocale(localeProps)` instead to avoid ts complaints

File: src/UrlProcessor.ts [hunk 2]
```diff
@@ -35,16 +35,22 @@ export class UrlProcessor {
         }
 
         let normalUrl: string;
-        let result: string;
         if (this.isRelative(url)) {
+            // console.debug('UrlProcessor.process relative url: ' + url);
```
Comment (pmi) [inline src/UrlProcessor.ts:39]: There's a lot of commented out `console.debugs`, did you forget to remove them ?

Comment (ComLock) [reply src/UrlProcessor.ts:39]: Nah, I can remove them, I just like keeping such for future debugging, but yah, that's just me.

File: src/UrlProcessor.ts [hunk 3]
```diff
@@ -70,21 +76,40 @@ export class UrlProcessor {
...
-        return linkData.find(data => data.ref === ref)?.media !== null;
+        if (!linkData || !Array.isArray(linkData) || !linkData.length) {
+            return false;
+        }
+        // Assuming linkData can't contain more than one entry with the same ref
+        const found = linkData.find(data => data?.ref === ref);
+        if (!found) {
+            return false;
+        }
+        return (found?.media || null) !== null;
```
[the remark this answers is not in the document]
Comment (ComLock) [reply src/UrlProcessor.ts]: Implemented :)

Now that I had tests for "every" possible input to the function, it was easy to check that the shortened, but not so readable code, worked.

Readability being a subjective measurement.

**Review by pmi — APPROVED** [review]

**Pull request merged**